app/src/main/java/com/nervousfish/nervousfish/activities/RhythmCreateActivity.java

```diff
@@ -0,0 +1,173 @@
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class RhythmCreateActivity extends AppCompatActivity {
+    private static final Logger LOGGER = LoggerFactory.getLogger("RhythmCreateActivity");
+    private ArrayList<SingleTap> tapCombination;
+    private IServiceLocator serviceLocator;
+    //TODO: change contact into an encrypted string of bytes
+    private Contact dataReceived = null;
+
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_rhythm_create);
+        final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_create_rhythm);
+        setSupportActionBar(toolbar);
+
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayShowTitleEnabled(false);
+        }
+
+        final Intent intent = this.getIntent();
+        this.serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+
+        LOGGER.info("RhythmActivity started");
+    }
+
+    /**
+     * Gets triggered when the tap screen is clicked.
+     *
+     * @param v - the {@link View} clicked
+     */
+    public void onTapClick(final View v) {
+        LOGGER.info("Tapped");
+        final Button startButton = (Button) this.findViewById(R.id.start_recording_button);
+        if (tapCombination != null && startButton.getVisibility() == View.GONE) {
+            tapCombination.add(new SingleTap(new Timestamp(System.currentTimeMillis())));
+        }
+    }
+
+    /**
+     * Gets triggered when the done button is clicked.
+     *
+     * @param v - the {@link View} clicked
+     */
+    public void onDoneClick(final View v) {
+        LOGGER.info("Done tapping button clicked");
+        try {
+            this.serviceLocator.getBluetoothHandler().send(new Contact("Baas", new SimpleKey("testkey", "456um4h692406u2p")));
+        } catch (IOException e) {
+            LOGGER.error("Could not send my contact to other device " + e.getMessage());
+        }
+        final Intent intent = new Intent(this, WaitActivity.class);
+        intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, serviceLocator);
+        intent.putExtra(ConstantKeywords.WAIT_MESSAGE, this.getString(R.string.wait_message_partner_rhythm_tapping));
+        intent.putExtra(ConstantKeywords.DATA_RECEIVED, dataReceived);
+        intent.putExtra(ConstantKeywords.TAP_DATA, tapCombination);
+        this.startActivityForResult(intent, ConstantKeywords.START_RHYTHM_REQUEST_CODE);
+    }
+
+    /**
+     * Gets triggered when the start recording button is clicked.
+     *
+     * @param v - the {@link View} clicked
+     */
+    public void onStartRecordingClick(final View v) {
+        LOGGER.info("Start Recording clicked");
+        tapCombination = new ArrayList<>();
+
+        final Button startButton = (Button) this.findViewById(R.id.start_recording_button);
```

Comment:
Could you move these assignment to onCreate (still to final variables though) because you now have several times the same code (also in onSteopRecordingClick)